pkg/cloud/services/ec2/dedicatedhosts.go

```diff
@@ -0,0 +1,260 @@
+	input := &ec2.AllocateHostsInput{
+		InstanceFamily:   aws.String(spec.InstanceFamily),
+		AvailabilityZone: aws.String(targetAZ),
+		Quantity:         aws.Int32(quantity),
+	}
+
+	// Set instance type if specified
+	if instanceType != "" {
+		input.InstanceType = aws.String(instanceType)
+	}
+
+	// Add tags if specified
+	if len(spec.Tags) > 0 {
+		var tagSpecs []types.TagSpecification
+		var tags []types.Tag
+		for key, value := range spec.Tags {
+			tags = append(tags, types.Tag{
+				Key:   aws.String(key),
+				Value: aws.String(value),
+			})
+		}
+		tagSpecs = append(tagSpecs, types.TagSpecification{
+			ResourceType: types.ResourceTypeDedicatedHost,
+			Tags:         tags,
+		})
+		input.TagSpecifications = tagSpecs
+	}
+
+	output, err := s.EC2Client.AllocateHosts(ctx, input)
+	if err != nil {
+		record.Warnf(s.scope.InfraCluster(), "FailedAllocateDedicatedHost", "Failed to allocate dedicated host: %v", err)
+		return "", errors.Wrap(err, "failed to allocate dedicated host")
+	}
+
+	if len(output.HostIds) == 0 {
+		return "", errors.New("no dedicated host ID returned from allocation")
+	}
+
+	hostID := output.HostIds[0]
+	s.scope.Info("Successfully allocated dedicated host", "hostID", hostID, "availabilityZone", targetAZ)
+	record.Eventf(s.scope.InfraCluster(), "SuccessfulAllocateDedicatedHost", "Allocated dedicated host %s in %s", hostID, targetAZ)
+
+	return hostID, nil
+}
+
+// ReleaseDedicatedHost releases a dedicated host.
+func (s *Service) ReleaseDedicatedHost(ctx context.Context, hostID string) error {
+	s.scope.Debug("Releasing dedicated host", "hostID", hostID)
+
+	input := &ec2.ReleaseHostsInput{
+		HostIds: []string{hostID},
+	}
+
+	_, err := s.EC2Client.ReleaseHosts(ctx, input)
+	if err != nil {
+		record.Warnf(s.scope.InfraCluster(), "FailedReleaseDedicatedHost", "Failed to release dedicated host %s: %v", hostID, err)
+		return errors.Wrap(err, "failed to release dedicated host")
+	}
+
+	s.scope.Info("Successfully released dedicated host", "hostID", hostID)
+	record.Eventf(s.scope.InfraCluster(), "SuccessfulReleaseDedicatedHost", "Released dedicated host %s", hostID)
+
+	return nil
+}
+
+// DescribeDedicatedHost describes a specific dedicated host.
+func (s *Service) DescribeDedicatedHost(ctx context.Context, hostID string) (*infrav1.DedicatedHostInfo, error) {
+	s.scope.Debug("Describing dedicated host", "hostID", hostID)
+
+	input := &ec2.DescribeHostsInput{
+		HostIds: []string{hostID},
+	}
+
+	output, err := s.EC2Client.DescribeHosts(ctx, input)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to describe dedicated host")
+	}
+
+	if len(output.Hosts) == 0 {
+		return nil, errors.Errorf("dedicated host %s not found", hostID)
+	}
+
+	host := output.Hosts[0]
+	hostInfo := s.convertToHostInfo(host)
+
+	return hostInfo, nil
+}
+
+// ListDedicatedHosts lists dedicated hosts based on filters.
+func (s *Service) ListDedicatedHosts(ctx context.Context, filters map[string]string) ([]*infrav1.DedicatedHostInfo, error) {
```

Comment:
where is this used?

Reply:
it was a vestige from initial work on this, it is not used and has been removed.